src/overlays/actors/ovl_Obj_Warpstone/z_obj_warpstone.h

```diff
@@ -5,14 +5,30 @@
 
 struct ObjWarpstone;
 
-typedef void (*ObjWarpstoneActionFunc)(struct ObjWarpstone* this, GlobalContext* globalCtx);
+typedef s32 (*ObjWarpstoneActionFunc)(struct ObjWarpstone* this, GlobalContext* globalCtx);
+
+typedef enum {
+    /* 0 */ SEK_MODEL_CLOSED,
+    /* 1 */ SEK_MODEL_OPENED
+} ObjectSekModels;
+
+typedef enum {
+    /* 25 */ OBJ_WARPSTONE_TIMER_OPENED = 25,
+    /* 66 */ OBJ_WARPSTONE_TIMER_DONE = 66
+} ObjWarpstoneTimerCutoffs;
 
 typedef struct ObjWarpstone {
-    /* 0x0000 */ Actor actor;
-    /* 0x0144 */ char unk_144[0x68];
-    /* 0x01AC */ ObjWarpstoneActionFunc actionFunc;
+    /* 0x000 */ DynaPolyActor dyna;
+    /* 0x15C */ ColliderCylinder collider; 
+    /* 0x1A8 */ u8 isTalking;
+    /* 0x1A9 */ u8 timer /*for statue opening cutscene*/;
+    /* 0x1AA */ u8 modelIndex;
+    /* 0x1AC */ ObjWarpstoneActionFunc actionFunc;
 } ObjWarpstone; // size = 0x1B0
 
 extern const ActorInit Obj_Warpstone_InitVars;
 
+#define GET_OWL_ID(this) ((u16)(this->dyna.actor.params & 0xF))
+#define IS_OWL_HIT(owlId) (((void)0,gSaveContext.owlsHit) & (u16)gBitFlags[owlId])
```

Comment:
How will these MACRO names be distinguished from the params used in `En_Owl` for Kaepora?

Reply:
Well it's based on the owlsHit bitfield in the save context. Should I rename that to warpstonesHit as well?

Reply:
I think a better solution would just be to use `OWL_STATUE` as the designator of this actor's internal enums/defines.

Reply:
submitting a new PR since this one got merged before I could get to addressing this comment. That PR incorporates this pseudo "namespacing" for lack of a better word.